src/ekore/anomalous_dimensions/unpolarized/time_like/as3.py

```diff
@@ -0,0 +1,832 @@
+    NI2 = NI * NI
+    NI3 = NI * NI2
+    NM = N - 1
+    NMI = 1 / NM
+    NMI2 = NMI * NMI
+    NMI3 = NMI * NMI2
+    N1 = N + 1
+    N1I = 1 / N1
+    N1I2 = N1I * N1I
+    N1I3 = N1I * N1I2
+    N2 = N + 2
+    N2I = 1 / N2
+    N2I2 = N2I * N2I
+    N2I3 = N2I * N2I2
+    S1 = w1.S1(N)  # c.get(c.S1, cache, N, is_singlet)
+    S2 = w2.S2(N)  # c.get(c.S2, cache, N, is_singlet)
+    S3 = w3.S3(N)  # c.get(c.S3, cache, N, is_singlet)
+    S4 = w4.S4(N)  # c.get(c.S4, cache, N, is_singlet)
+    S1M = S1 - NI
+    S11 = S1 + N1I
+    S12 = S11 + N2I
+    S2M = S2 - NI2
+    S21 = S2 + N1I2
+    S22 = S21 + N2I2
+    S31 = S3 + N1I3
+    A0 = -S1M
+    B1 = -S1 * NI
+    C0 = NI
+    CM = NMI
+    C1 = N1I
+    C2 = N2I
+    C3 = 1 / (N + 3)
+    C4 = 1 / (N + 4)
+    D1 = -NI2
+    D1M = -NMI2
+    D2 = 2 * NI3
+    D2M = 2 * NMI3
+    D3 = -6 * NI2 * NI2
+    D3M = -6 * NMI2 * NMI2
+    D31 = -6 * N1I2 * N1I2
+    D4 = 24 * NI2 * NI3
+    D4M = 24 * NMI2 * NMI3
+    E1 = S1 * NI2 + (S2 - zeta2) * NI
+    E11 = S11 * N1I2 + (S21 - zeta2) * N1I
+    E2 = 2 * (-S1 * NI3 + (zeta2 - S2) * NI2 - (S3 - zeta3) * NI)
+    F1 = 2 * NI * (zeta3 + zeta2 * S1 - 0.5 * NI * (S1 * S1 + S2) - S1 * S2 - S3)
+
+    GG0 = (
+        576 * D4M
+        + 3168 * D3M
+        + 3651.1 * D2M
+        + 10233 * D1M
+        + 14214.4 * CM
+        + 191.99 * D4
+        + 3281.7 * D3
+        + 13528 * D2
+        + 12258 * D1
+        - 28489 * C0
+        + 7469 * C1
+        + 30421 * C2
+        - 53017 * C3
+        + 19556 * C4
+        - 186.4 * E1
+        - 21328 * E2
+        + 5685.8 * D31
+        - 3590.1 * B1
+        + 4425.451
+        + 2643.521 * A0
+    )
+    GG1 = (
+        448 / 9 * D3M
+        + 2368 / 9 * D2M
+        - 5.470 * D1M
+        - 804.13 * CM
+        + 18.085 * D4
+        + 155.10 * D3
+        + 482.94 * D2
+        + 4.9934 * D1
+        + 248.95 * C0
+        + 260.6 * C1
+        + 272.79 * C2
+        + 2133.2 * C3
+        - 926.87 * C4
+        + 1266.5 * E1
+        - 29.709 * E2
+        + 87.771 * F1
+        + 485.18 * D31
+        + 319.97 * B1
+        - 528.719
+        - 412.172 * A0
+    )
+    GG2 = (
+        32 / 27 * D2M
+        + 368 / 81 * D1M
+        + 472 / 243 * CM
+        - 5.0372 * D3
+        - 44.80 * D2
+        - 69.712 * D1
+        - 77.190 * C0
+        + 153.27 * C1
+        - 106.03 * C2
+        + 11.995 * C3
+        - 115.01 * E1
+        + 96.522 * E11
+        - 62.908 * E2
+        + 6.4628
+        - 16 / 9 * A0
+    )
+
+    result = GG0 + nf * (GG1 + nf * GG2)
+    return -result
+
+
+@nb.njit(cache=True)
+def gamma_singlet(N, nf, cache, is_singlet=True):
+    r"""Compute the NNLO singlet anomalous dimension matrix.
+
+    Parameters
+    ----------
+    N : complex
+        Mellin moment
+    nf : int
+        No. of active flavors
+    cache : numpy.ndarray
+        Harmonic sum cache
+    is_singlet : boolean
+        True for singlet, False for non-singlet, None otherwise
```

Comment:
If it were actually three-valued, you should have written:
```suggestion
    is_singlet : boolean or None
        True for singlet, False for non-singlet, None otherwise
```

But, as @giacomomagni wrote above, `None` makes no sense, and, as I wrote above, here even `False` makes no sense: it is a singlet quantity.